jul-ecs-formatter/src/main/java/co/elastic/logging/jul/EcsFormatter.java

```diff
@@ -109,9 +109,19 @@ public void setEventDataset(String eventDataset) {
     }
 
     public void setAdditionalFields(String additionalFields) {
+        if (!this.additionalFields.isEmpty()) {
+            throw new IllegalStateException("Additional fields have already been set");
+        }
```

Comment:
As it's a setter, just from reading the method name, I'd intuitively expect it to just override any previous state rather than throwing an exception.

Reply:
Since it's extremely unlikely that someone tries to set twice, I'd assume it is a mistake that I rather tell the user about.
However, I could say the same about setting twice through the same setter, so I agree, this should be removed